packages/service-library/src/servicelib/prometheus_metrics.py

```diff
@@ -0,0 +1,220 @@
+# # HELP python_gc_objects_uncollectable_total Uncollectable object found during GC
+# # TYPE python_gc_objects_uncollectable_total counter
+# python_gc_objects_uncollectable_total{generation="0"} 0.0
+# python_gc_objects_uncollectable_total{generation="1"} 0.0
+# python_gc_objects_uncollectable_total{generation="2"} 0.0
+# # HELP python_gc_collections_total Number of times this generation was collected
+# # TYPE python_gc_collections_total counter
+# python_gc_collections_total{generation="0"} 628.0
+# python_gc_collections_total{generation="1"} 57.0
+# python_gc_collections_total{generation="2"} 5.0
+# # HELP http_requests_total Total requests count
+# # TYPE http_requests_total counter
+# http_requests_total{app_name="simcore_service_webserver",endpoint="/v0/",http_status="200",method="GET"} 15.0
+# # HELP http_requests_created Total requests count
+# # TYPE http_requests_created gauge
+# http_requests_created{app_name="simcore_service_webserver",endpoint="/v0/",http_status="200",method="GET"} 1.6418063614890063e+09
+# # HELP http_in_flight_requests Number of requests in process
+# # TYPE http_in_flight_requests gauge
+# http_in_flight_requests{app_name="simcore_service_webserver",endpoint="/v0/",method="GET"} 0.0
+# http_in_flight_requests{app_name="simcore_service_webserver",endpoint="/metrics",method="GET"} 1.0
+# # HELP http_request_latency_seconds Time processing a request
+# # TYPE http_request_latency_seconds summary
+# http_request_latency_seconds_count{app_name="simcore_service_webserver",endpoint="/v0/",method="GET"} 15.0
+# http_request_latency_seconds_sum{app_name="simcore_service_webserver",endpoint="/v0/",method="GET"} 0.007384857000033662
+# http_request_latency_seconds_count{app_name="simcore_service_webserver",endpoint="/metrics",method="GET"} 0.0
+# http_request_latency_seconds_sum{app_name="simcore_service_webserver",endpoint="/metrics",method="GET"} 0.0
+# # HELP http_request_latency_seconds_created Time processing a request
+# # TYPE http_request_latency_seconds_created gauge
+# http_request_latency_seconds_created{app_name="simcore_service_webserver",endpoint="/v0/",method="GET"} 1.6418063614873598e+09
+# http_request_latency_seconds_created{app_name="simcore_service_webserver",endpoint="/metrics",method="GET"} 1.641806371709292e+09
+
+
+@dataclass
+class PrometheusMetrics:
+    registry: CollectorRegistry
+    process_collector: ProcessCollector
+    platform_collector: PlatformCollector
+    gc_collector: GCCollector
+    request_count: Counter
+    in_flight_requests: Gauge
+    response_latency_with_labels: Histogram
+    response_latency_detailed_buckets: Histogram
+
+
+def _get_exemplar() -> dict[str, str] | None:
+    current_span = trace.get_current_span()
+    if not current_span.is_recording():
+        return None
+    trace_id = trace.format_trace_id(current_span.get_span_context().trace_id)
+    return {"TraceID": trace_id}
+
+
+def setup_prometheus_metrics() -> PrometheusMetrics:
```

Comment:
setup name is a bit special. we now have `setup` and `initialize` and `lifespan` which would get mixed up with this one. Since you are not setting a module in fastapi please consider the below rename or something else that does not conclude the word "setup".
```suggestion
def get_prometheus_metrics() -> PrometheusMetrics:
```